app/controllers/transactions_controller.rb

```diff
@@ -57,46 +36,21 @@ def index
 			}
 			@all_transactions.push(resp)
 		end
-
 	end
 
 	def dashboard
-		@total_rev = 0
-		@daily_sales = [0, 0, 0, 0, 0, 0, 0]
-		@hourly_sales = {}
-		@movie_sales = []
-
-		Transaction.all.each do |transaction|
-			# Add the total from the transaction to total_rev
-			@total_rev += transaction.cost
-			# Find the showing the transaction belongs to
-			showing = transaction.showing
-			# Add the total from the transaction to showings day of the week
-			@daily_sales[showing.date.wday] += transaction.cost
+		dash_services = DashboardService
+		@total_rev = dash_services.total_rev
+		@daily_sales = dash_services.daily_sales
+		@hourly_sales = dash_services.hourly_sales
+		@movie_sales = dash_services.movie_sales
+	end
 
-			# Check to see if the show time has been see already - if so add the transaction cost to the total
-			@hourly_sales.has_key?(showing.time) ? @hourly_sales[showing.time] += transaction.cost : 
-				# Otherwise set the total equal to the transaction cost
-				@hourly_sales[showing.time] = transaction.cost
-		end #End Transaction.all.each loop
 
-		# Find each movies total revenue
-		Movie.all.each do |movie|
-			movie_sales = 0
+	private
 
-			# Loop through all of the transactions for each movie
-			movie.transactions.each do |transaction|
-				movie_sales += transaction.cost
-			end
-			
-			movie = {
-				id: movie.id,
-				title: movie.title,
-				sales: movie_sales
-			}
-			@movie_sales.push(movie)
-		end #End Movie.all.each loop
-
-	end #End Dashboard Route
+	def transaction_params
+		params.require(:transaction).permit(:email, :first_name, :last_name, :showing_id, :quantity)
```

Comment:
Nice! If you wanted, you could also get some practice using `Decanter` by adding the gem, and a `TransactionDecanter`, and here you would return `@transaction_params = TransactionDecanter.decant(params[:transaction])`